tests/classification/test_auroc.py

```diff
@@ -86,18 +86,18 @@ def _sk_auroc_multilabel_multidim_prob(preds, target, num_classes, average='macr
     )
 
 
-@pytest.mark.parametrize(
-    "preds, target, sk_metric, num_classes",
-    [(_input_binary_prob.preds, _input_binary_prob.target, _sk_auroc_binary_prob, 1),
-     (_input_mcls_prob.preds, _input_mcls_prob.target, _sk_auroc_multiclass_prob, NUM_CLASSES),
-     (_input_mdmc_prob.preds, _input_mdmc_prob.target, _sk_auroc_multidim_multiclass_prob, NUM_CLASSES),
-     (_input_mlb_prob.preds, _input_mlb_prob.target, _sk_auroc_multilabel_prob, NUM_CLASSES),
-     (_input_mlmd_prob.preds, _input_mlmd_prob.target, _sk_auroc_multilabel_multidim_prob, NUM_CLASSES)]
-)
 @pytest.mark.parametrize("average", ['macro', 'weighted', 'micro'])
 @pytest.mark.parametrize("max_fpr", [None, 0.8, 0.5])
 class TestAUROC(MetricTester):
 
+    @pytest.mark.parametrize(
+        "preds, target, sk_metric, num_classes", [
+            (_input_binary_prob.preds, _input_binary_prob.target, _sk_auroc_binary_prob, 1),
+            (_input_mcls_prob.preds, _input_mcls_prob.target, _sk_auroc_multiclass_prob, NUM_CLASSES),
+            (_input_mdmc_prob.preds, _input_mdmc_prob.target, _sk_auroc_multidim_multiclass_prob, NUM_CLASSES),
+            (_input_mlb_prob.preds, _input_mlb_prob.target, _sk_auroc_multilabel_prob, NUM_CLASSES),
+            (_input_mlmd_prob.preds, _input_mlmd_prob.target, _sk_auroc_multilabel_multidim_prob, NUM_CLASSES)]
+    )
```

Comment:
move this in front of the class so it parametrize all tests and remove all duplicated
even though `test_auroc_differentiability` does not use the `sk_metric` arg it can still be part of the function definition

Reply:
Sure, I will do these changes!